### PR TITLE
Content sync and updated to md | global_objects/asyncfunction [ES]

### LINKED ISSUE #4163: 3. web/javascript/reference/global_objects/asyncfunction/index.html

### PR DESCRIPTION
**Pull request by Vallejoanderson**
Fixed #4163

### CODE REVIEW

**Comment by SphinxKnight** [comment]
(failure for AutoCorrect is due to https://github.com/mdn/translated-content/issues/4170 and not due to this PR content, feel free to ignore it or to rebase this branch against a more recent state to avoid this workflow running)

**Comment by Graywolf9** [comment]
> (failure for AutoCorrect is due to #4170 and not due to this PR content, feel free to ignore it or to rebase this branch against a more recent state to avoid this workflow running)

Thanks for the comment !

File: files/es/web/javascript/reference/global_objects/asyncfunction/index.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+new AsyncFunction(arg0, functionBody)
+new AsyncFunction(arg0, arg1, functionBody)
+new AsyncFunction(arg0, arg1, ...argN, functionBody)
+```
+
+### Parámetros
+
+- `arg1, arg2, ... argN`
+
+  - : Nombres a ser usados por la función como los nombres de los argumentos. Cada uno debe ser una cadena que corresponda a un identificador válido en JavaScript (cualquier [identificador](/es/docs/Glossary/Identifier), [parámetro rest](/es/docs/Web/JavaScript/Reference/Functions/rest_parameters), o [parámetro destructurado](/es/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment), opcionalmente con un valor por defecto), o una lista de tales cadenas separadas por comas.
```
Comment (Graywolf9) [inline files/es/web/javascript/reference/global_objects/asyncfunction/index.md:31]:
```suggestion
  - : Nombres a ser usados por la función como los nombres de los argumentos. Cada uno debe ser una cadena que corresponda a un identificador válido en JavaScript (cualquier [identificador](/es/docs/Glossary/Identifier), [parámetro rest](/es/docs/Web/JavaScript/Reference/Functions/rest_parameters), o [parámetro desestructurado](/es/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment), opcionalmente con un valor por defecto), o una lista de tales cadenas separadas por comas.
```

File: files/es/web/javascript/reference/global_objects/asyncfunction/index.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+    Así mismo, los parámetros se procesan de la misma manera que las declaraciones de funciones, aceptando espacios y comentarios. Por ejemplo: `"x", "theValue = 42", "[a, b] /* numbers */"` - o `"x, theValue = 42, [a, b] /* numbers */"`. (`"x, theValue = 42", "[a, b]"` es también correcto, pero muy confuso para leer).
```
Comment (Graywolf9) [inline files/es/web/javascript/reference/global_objects/asyncfunction/index.md:33]:
```suggestion
    Así mismo, los parámetros se procesan de la misma manera que las declaraciones de funciones, aceptando espacios y comentarios. Por ejemplo: `"x", "theValue = 42", "[a, b] /* numbers */"` — o `"x, theValue = 42, [a, b] /* numbers */"`. (`"x, theValue = 42", "[a, b]"` es también correcto, pero muy confuso para leer).
```

File: files/es/web/javascript/reference/global_objects/asyncfunction/index.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+  - : `functionBody`
```
Comment (Graywolf9) [inline files/es/web/javascript/reference/global_objects/asyncfunction/index.md:35]:
```suggestion
- : `functionBody`
```

File: files/es/web/javascript/reference/global_objects/asyncfunction/index.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    - : Una cadena que contiene las sentencias de JavaScript que componen la definición de la función.
```
Comment (Graywolf9) [inline files/es/web/javascript/reference/global_objects/asyncfunction/index.md:36]:
```suggestion
  - : Una cadena que contiene las sentencias de JavaScript que componen la definición de la función.
```

File: files/es/web/javascript/reference/global_objects/asyncfunction/index.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+## Descripción
+
+Los objetos {{jsxref("Statements/async_function", "async function")}} creados con el constructor `AsyncFunction` son procesados en el momento que la función es creada. Esto es menos eficiente que declarar una función asincrona con un {{jsxref("Statements/async_function", "async function expression")}} y llamarla con eso en tu código , ya que las funciones están procesadas junto al resto del código.
```
Comment (Graywolf9) [inline files/es/web/javascript/reference/global_objects/asyncfunction/index.md:40]:
```suggestion
Los objetos {{jsxref("Statements/async_function", "async function")}} creados con el constructor `AsyncFunction` son procesados en el momento que la función es creada. Esto es menos eficiente que declarar una función asincrona con un {{jsxref("Statements/async_function", "async function expression")}} y llamarla con eso en tu código , ya que las funciones son procesadas junto al resto del código.
```

File: files/es/web/javascript/reference/global_objects/asyncfunction/index.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+Todos los argumentos que son pasados a la función son tratados por los nombres de los identificadores de los parámetros en la función creada, en el orden en que son pasados a la función.
+
+> **Nota:** Las {{jsxref("Statements/async_function", "async functions", "", 1)}} creadas con el constructor `AsyncFunction` no crean 
+> closures en sus contextos creados, siempre son creados en el contexto global.
```
Comment (Graywolf9) [inline files/es/web/javascript/reference/global_objects/asyncfunction/index.md:45]:
```suggestion
> [_closures_](/es/docs/Web/JavaScript/Closures) en sus contextos creados, siempre son creados en el contexto global.
```

Para tener una referencia y qué es la palabra no traducida

File: files/es/web/javascript/reference/global_objects/asyncfunction/index.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+let AsyncFunction = Object.getPrototypeOf(async function(){}).constructor
+
+let a = new AsyncFunction('a',
+                          'b',
+                          'return await resolveAfter2Seconds(a) + await resolveAfter2Seconds(b);');
+
+a(10, 20).then(v => {
+  console.log(v); // imprimir 30 después de 4 segundos
```
Comment (Graywolf9) [inline files/es/web/javascript/reference/global_objects/asyncfunction/index.md:75]:
```suggestion
  console.log(v); // imprime 30 después de 4 segundos
```

sugerencia

**Review by Graywolf9 — CHANGES_REQUESTED** [review]

File: files/es/web/javascript/reference/global_objects/asyncfunction/index.md (hunk 6 above)
Comment (Vallejoanderson) [reply files/es/web/javascript/reference/global_objects/asyncfunction/index.md:45]: Gracias, @Graywolf9

**Review by Graywolf9 — APPROVED** [review]

**Pull request merged**